src/trigrams.rs

```diff
@@ -1,6 +1,6 @@
 use std::collections::HashMap;
 
-pub fn get_trigrams_with_positions(text : &String) -> HashMap<String, u32> {
+pub fn get_trigrams_with_positions(text : &str) -> HashMap<String, u32> {
```

Comment:
It should take a `&str`. Rust makes borrowing (and, by omission of `&`, moves) explicit. I don't think there's any reason a query would need to take ownership of the text, so I think you should stick to a borrow.

There's no real reason to ever use `&String` over `&str`. Everywhere a `&str` is expected, you can always pass a `&String`. And then you can change lines like [this code](https://github.com/greyblake/whatlang-rs/blob/2dd805866ddd570912088f2f16ef556a2b827119/src/query.rs#L38) to be

```rust
let query = Query::new("Testing");
```